pkg/bisect/bisect.go

```diff
@@ -348,6 +348,22 @@ func (env *env) identifyRewrittenCommit() (string, error) {
 		return cfg.Kernel.Commit, err
 	}
 
+	if !cfg.Fix {
+		// If we're doing a cause bisection, we don't really need the commit to be
+		// reachable from cfg.Kernel.Branch.
+		// So let's try to force tag fetch and check if the commit is present in the
+		// repository.
+		env.log("fetch other tags and check if the commit is present")
+		commit, err := env.repo.CheckoutCommit(cfg.Kernel.Repo, cfg.Kernel.Commit)
+		if err != nil {
+			// Ignore the error because the command will fail if the commit is really not
+			// present in the tree.
+			env.log("fetch failed with %s", err)
+		} else if commit != nil {
+			return commit.Hash, nil
```

Comment:
It's still the same commit, not really rewritten. I would just fetch the tree before besiction in a way that makes commits available. Then we don't need any special code for this case.

Can we now remove the commit title logic? You said it resolves what this commit is supposed to resolve.

At the start of the function we do CheckoutBranch, but as far I see we already did it before bisection. So potentially we can remove this whole function if we fetch the tree properly.

Reply:
The code mentioned bewlo another use case for commit titles -- a downstream kernel that gets rebased over the upstream one. I think it makes sense to keep `pkg/bisect` support this.

> Then we don't need any special code for this case.

We still need to somewhat specialcase here -- for fix bisections, we do want to ensure that the commit is reachable from the branch's HEAD.

> I would just fetch the tree before besiction in a way that makes commits available. 

But it currently lets us save some resources by only checking what is necessary. For non-linux-next bisections, we only need to check out what's reachable from the branch of interest. But if we do `--tags` for all checkouts, we'd force it to fetch more stuff, even when it's not needed.